src/main.nr

```diff
@@ -93,17 +106,31 @@ contract PrivateOracle {
     }
  
     // Constructs the contract.
-    #[aztec(private)] 
-    fn constructor() {
+    #[aztec(private)]
+    fn constructor(payment_token: AztecAddress, fee: Field) {
+        let selector = compute_selector("_initialize((Field),Field)");
+        context.call_public_function(context.this_address(), selector, [payment_token.address,fee]);
     }
 
     // Requester submit a question
     #[aztec(private)]
-    fn submit_question(question: Field, divinity_address: AztecAddress) {
-        let storage = Storage::init(Context::private(&mut context));
+    fn submit_question(question: Field, divinity_address: AztecAddress, nonce: Field) {
+        let payment_token = payment_token();
```

Comment:
payment_token() and fee() are unconstrained -> any witness for them will create a valid proof (it's even worse in cancel question, as you can then withdraw whatever amount with a cancellation)

-> use singleton (so we can access them in private fn) and unconstrained getters for the outside world (exposing them)?